Don't use backslashes when UTF-8 mode is on.
Index: html2text-1.3.2a/html2text.C
===================================================================
--- html2text-1.3.2a.orig/html2text.C	2008-09-13 08:56:22.008960077 +0300
+++ html2text-1.3.2a/html2text.C	2008-09-13 08:57:03.120443520 +0300
@@ -212,6 +212,10 @@
       exit(1);
     }
   }
+  if (use_encoding == UTF8)
+  {
+	  use_backspaces = false;
+  }
   if (i > argc) {
     std::cerr
       << "Error: Required parameter after \""
